plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoCaseInsensitiveNameMatchingDisabled.java

```diff
@@ -0,0 +1,102 @@
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+import static io.trino.plugin.mongodb.MongoQueryRunner.createMongoClient;
+import static io.trino.plugin.mongodb.MongoQueryRunner.createMongoQueryRunner;
+import static io.trino.testing.TestingNames.randomNameSuffix;
+import static java.util.Locale.ENGLISH;
+
+public class TestMongoCaseInsensitiveNameMatchingDisabled
+        extends AbstractTestQueryFramework
+{
+    private MongoServer server;
+    private MongoClient client;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        server = new MongoServer();
+        client = createMongoClient(server);
+        return createMongoQueryRunner(server,
+                ImmutableMap.of(),
+                ImmutableMap.of(),
+                ImmutableMap.of("mongodb.case-insensitive-name-matching", "false"),
+                ImmutableList.of(),
+                (runner) -> {});
+    }
+
+    @AfterClass(alwaysRun = true)
+    public final void destroy()
+    {
+        server.close();
+        server = null;
+        client.close();
+        client = null;
+    }
+
+    @Test
+    public void testDescribeTable()
+    {
+        String schemaName = "test" + randomNameSuffix();
+        String tableName = "Test" + randomNameSuffix();
+        String schemaTableName = schemaName + "." + tableName.toLowerCase(ENGLISH);
+        client.getDatabase(schemaName).getCollection(tableName).insertOne(new Document("test", "aaa"));
+
+        assertQueryFails(
+                "DESC mongodb." + schemaTableName,
+                "line 1:1: Table 'mongodb." + schemaTableName + "' does not exist");
+    }
+
+    @Test
+    public void testListTables()
+    {
+        List<String> lowerCasedCollections = ImmutableList.of("collection1", "collection2");
+        List<String> upperCasedCollections = ImmutableList.of("Collection3", "COLLECTION4");
+        List<String> allCollections = ImmutableList.<String>builder().addAll(lowerCasedCollections).addAll(upperCasedCollections).build();
+        String schemaName = "test";
+
+        for (String collection : allCollections) {
+            client.getDatabase(schemaName).getCollection(collection).insertOne(new Document("test", "aaa"));
+        }
+
+        assertQuery("SHOW TABLES IN mongodb." + schemaName,
+                "VALUES ('collection1'), ('collection2')");
+    }
+
+    @Test
+    public void testSelectAllWithCaseInsensitive()
+    {
+        String schemaName = "test" + randomNameSuffix();
+        String tableName = "Test" + randomNameSuffix();
+        String schemaTableName = schemaName + "." + tableName.toLowerCase(ENGLISH);
+        client.getDatabase(schemaName).getCollection(tableName).insertOne(new Document("test", "aaa"));
+
+        assertQueryFails(
+                "SELECT * FROM mongodb." + schemaTableName,
+                "line 1:15: Table 'mongodb." + schemaTableName + "' does not exist");
+    }
```

Comment:
These test cases verify non-lowercase table name and they don't cover schema name case sensitivities. Can we add more test cases?

Reply:
I add tests with non-lowercase schema name.

Before, there was no implementation of schema name existence, so I add `schemaExists` and check schema existence in `loadTableSchema`.